Lib/_pyio.py

```diff
@@ -2458,6 +2458,8 @@ def detach(self):
         return buffer
 
     def seek(self, cookie, whence=0):
+        self._checkClosed()
```

Comment:
Can you move this one after `def _reset_encoder`? The diff would be easier to read (we don't really care that the function def is before or after the check IMO, and it's clearer to declare the local functions before the actual body)